cmake_unofficial/CMakeLists.txt

```diff
@@ -1,40 +1,99 @@
-cmake_minimum_required(VERSION 2.6)
-cmake_policy(VERSION 2.6)
+# To the extent possible under law, the author(s) have dedicated all
+# copyright and related and neighboring rights to this software to
+# the public domain worldwide. This software is distributed without
+# any warranty.
+#
+# For details, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 
-project(xxhash)
+set(XXHASH_TOP_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/..")
 
-set(XXHASH_LIB_VERSION "0.42.0")
-set(XXHASH_LIB_SOVERSION "0")
+# Parse version information
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_MAJOR REGEX "^#define XXH_VERSION_MAJOR +([0-9]+) *$")
+string(REGEX REPLACE "^#define XXH_VERSION_MAJOR +([0-9]+) *$" "\\1" XXHASH_VERSION_MAJOR "${XXHASH_VERSION_MAJOR}")
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_MINOR REGEX "^#define XXH_VERSION_MINOR +([0-9]+) *$")
+string(REGEX REPLACE "^#define XXH_VERSION_MINOR +([0-9]+) *$" "\\1" XXHASH_VERSION_MINOR "${XXHASH_VERSION_MINOR}")
+file(STRINGS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h" XXHASH_VERSION_RELEASE REGEX "^#define XXH_VERSION_RELEASE +([0-9]+) *$")
+string(REGEX REPLACE "^#define XXH_VERSION_RELEASE +([0-9]+) *$" "\\1" XXHASH_VERSION_RELEASE "${XXHASH_VERSION_RELEASE}")
+set(XXHASH_VERSION_STRING "${XXHASH_VERSION_MAJOR}.${XXHASH_VERSION_MINOR}.${XXHASH_VERSION_RELEASE}")
+mark_as_advanced(XXHASH_VERSION_STRING XXHASH_VERSION_MAJOR XXHASH_VERSION_MINOR XXHASH_VERSION_RELEASE)
 
-option(BUILD_XXHSUM "Build the xxhsum binary" ON)
+set_directory_properties(PROPERTIES CMAKE_CONFIGURE_DEPENDS "${XXHASH_TOP_SOURCE_DIR}/xxhash.h")
 
-# Make CMake's RPATH handling not be insane. This suff has cmake set rpaths appropriately for
-# where things end up in the install tree. For some reason that's not the default:
-# https://cmake.org/Wiki/CMake_RPATH_handling
-SET(CMAKE_SKIP_BUILD_RPATH FALSE)
-SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+if("${CMAKE_VERSION}" VERSION_LESS "3.0")
+  project(XXHASH C)
+else()
+  cmake_policy (SET CMP0048 NEW)
+  project(XXHASH
+    VERSION ${XXHASH_VERSION_STRING}
+    LANGUAGES C)
+endif()
 
-# Where we search for shared libraries
-SET(CMAKE_INSTALL_RPATH "./lib")
+cmake_minimum_required (VERSION 2.8.12)
 
-# add the automatically determined parts of the RPATH
-# which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+# If XXHASH is being bundled in another project, we don't want to
+# install anything.  However, we want to let people override this, so
+# we'll use the XXHASH_BUNDLED_MODE variable to let them do that; just
+# set it to OFF in your project before you add_subdirectory(xxhash/contrib/cmake_unofficial).
+get_directory_property(XXHASH_PARENT_DIRECTORY PARENT_DIRECTORY)
```

Comment:
If you don't remember a reason and none has been documented, then do the change as it is much simpler to read and correct.

CMake is having a bad reputation because of things like that "I don't know why, I don't understand any of it, but it had to be done that way". Then people copy-paste that all the time and bad patterns propagate.